lib/jni/PrivacyGuard_jni.cpp

```diff
@@ -4,46 +4,24 @@
 //
 #include "JniConvertors.hpp"
 #include "modules/privacyguard/PrivacyGuard.hpp"
+#include "PrivacyGuardState.hpp"
 #include "WrapperLogManager.hpp"
 
 using namespace MAT;
 
-CommonDataContext GenerateCommonDataContextObject(JNIEnv* env,
-                                                  jstring domainName,
-                                                  jstring machineName,
-                                                  jstring userName,
-                                                  jstring userAlias,
-                                                  jobjectArray ipAddresses,
-                                                  jobjectArray languageIdentifiers,
-                                                  jobjectArray machineIds,
-                                                  jobjectArray outOfScopeIdentifiers)
-{
-    CommonDataContext cdc;
-    cdc.DomainName = JStringToStdString(env, domainName);
-    cdc.MachineName = JStringToStdString(env, machineName);
-    cdc.UserName = JStringToStdString(env, userName);
-    cdc.UserAlias = JStringToStdString(env, userAlias);
-    cdc.IpAddresses = ConvertJObjectArrayToStdStringVector(env, ipAddresses);
-    cdc.LanguageIdentifiers = ConvertJObjectArrayToStdStringVector(env, languageIdentifiers);
-    cdc.MachineIds = ConvertJObjectArrayToStdStringVector(env, machineIds);
-    cdc.OutOfScopeIdentifiers = ConvertJObjectArrayToStdStringVector(env, outOfScopeIdentifiers);
-    return cdc;
-}
-
 extern "C"
 {
-std::shared_ptr<PrivacyGuard> spPrivacyGuard;
-
 JNIEXPORT jboolean JNICALL
 Java_com_microsoft_applications_events_PrivacyGuard_nativeInitializePrivacyGuardWithoutCommonDataContext(
         JNIEnv *env, jclass /* this */, jlong iLoggerNativePtr) {
-    if (spPrivacyGuard != nullptr) {
+    if (PrivacyGuardState::isPrivacyGuardInstanceInitialized()) {
         return false;
     }
 
     InitializationConfiguration config;
     config.LoggerInstance = reinterpret_cast<ILogger*>(iLoggerNativePtr);
-    spPrivacyGuard = std::make_shared<PrivacyGuard>(config);
+    auto spPrivacyGuard = std::make_shared<PrivacyGuard>(config);
+    PrivacyGuardState::setPrivacyGuardInstance((spPrivacyGuard));
```

Comment:
Wouldn't you simply want a GetOrCreate style method?

Reply:
Ah good call, that is significantly cleaner.